packages/flutter/lib/src/rendering/flow.dart

```diff
@@ -0,0 +1,345 @@
+    else
+      child.parentData = new _FlowParentData();
+  }
+
+  /// The delegate that controls the transformation matrices of the children.
+  FlowDelegate get delegate => _delegate;
+  FlowDelegate _delegate;
+  void set delegate (FlowDelegate newDelegate) {
+    assert(newDelegate != null);
+    if (_delegate == newDelegate)
+      return;
+    final FlowDelegate oldDelegate = _delegate;
+    _delegate = newDelegate;
+
+    if (newDelegate.runtimeType != oldDelegate.runtimeType || newDelegate.shouldRelayout(oldDelegate))
+      markNeedsLayout();
+    else if (newDelegate.shouldRepaint(oldDelegate))
+      markNeedsPaint();
+
+    if (attached) {
+      oldDelegate._repaint?.removeListener(markNeedsPaint);
+      newDelegate._repaint?.addListener(markNeedsPaint);
+    }
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _delegate._repaint?.addListener(markNeedsPaint);
+  }
+
+  @override
+  void detach() {
+    _delegate._repaint?.removeListener(markNeedsPaint);
+    super.detach();
+  }
+
+  Size _getSize(BoxConstraints constraints) {
+    assert(constraints.debugAssertIsValid());
+    return constraints.constrain(_delegate.getSize(constraints));
+  }
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  @override
+  double getMinIntrinsicWidth(BoxConstraints constraints) {
+    return _getSize(constraints).width;
+  }
+
+  @override
+  double getMaxIntrinsicWidth(BoxConstraints constraints) {
+    return _getSize(constraints).width;
+  }
+
+  @override
+  double getMinIntrinsicHeight(BoxConstraints constraints) {
+    return _getSize(constraints).height;
+  }
+
+  @override
+  double getMaxIntrinsicHeight(BoxConstraints constraints) {
+    return _getSize(constraints).height;
+  }
+
+  @override
+  void performLayout() {
+    size = _getSize(constraints);
+    int i = 0;
+    RenderBox child = firstChild;
+    while (child != null) {
+      BoxConstraints innerConstraints = _delegate.getConstraintsForChild(i, constraints);
+      child.layout(innerConstraints, parentUsesSize: true);
+      final _FlowParentData childParentData = child.parentData;
+      childParentData.offset = Offset.zero;
+      child = childParentData.nextSibling;
+      i += 1;
+    }
+  }
+
+  final List<int> _lastPaintOrder = <int>[];
+  PaintingContext _paintingContext;
+  Offset _paintingOffset;
+  List<RenderBox> _randomAccessChildren;
+
+  @override
+  Size getChildSize(int i) {
+    if (i < 0 || i >= _randomAccessChildren.length)
+      return null;
+    return _randomAccessChildren[i].size;
+  }
+
+  @override
+  void paintChild(int i, Matrix4 transform) {
+    transform ??= new Matrix4.identity();
+    _lastPaintOrder.add(i);
+    RenderBox child = _randomAccessChildren[i];
+    final _FlowParentData childParentData = child.parentData;
+    assert(() {
+      if (childParentData._transform != null) {
+        throw new FlutterError(
+          'Paint each child at most once.\n'
```

Comment:
this will result in messages like:
```
Another exception was caught: Paint each child at most once.
Another exception was caught: Paint each child at most once.
Another exception was caught: Paint each child at most once.
```
I think it looks better when the first line (the one that gets extracted as an error message) reads like a more formal error message, and the helpful text is next (so that it appears in the long-form messages).